pupil_remote/Matlab_Python/pupilRead.m

```diff
@@ -0,0 +1,103 @@
+function [pupil_vals] = pupilRead(hUDP, eyeProperties)
+
+%process the packet from SURFACE Subscribe for Timeout
+pupil_vals.gazePosition = NaN([1,2],'single');
+pupil_vals.gazeConfidence = NaN(1,'single');
+pupil_vals.gazeTime = NaN(1,'single');
+pupil_vals.gazeOnSurface = NaN(1,'single');
+pupil_vals.pupilDiam = NaN(1,'single');
+pupil_vals.pupilDiamTime = NaN(1,'single');
+
+% return early if not connected
+if ~eyeProperties.isConnected,return;end
+
+% get data
+pkt = nan;
+num_iter = 0;
+max_iter = 20;
+fclose(hUDP);
+fopen(hUDP);
```

Comment:
If i remember correctly if you don't do this buffers will start to hold old pupil gaze information